modules/claims_api/app/controllers/claims_api/v1/forms/base.rb

```diff
@@ -61,7 +65,7 @@ def documents
         def decode_document(document)
           base64 = document.split(',').last
           decoded_data = Base64.decode64(base64)
-          filename = "temp_upload_#{Time.zone.now.to_i}.pdf"
+          filename = "temp_upload_#{SecureRandom.urlsafe_base64(8)}.pdf"
```

Comment:
While locally running through everything, I noticed that if the processing is fast enough, you can end up with 2 (or more) Base64 encoded attachments that get the same timestamp in milliseconds.  Which would lead to the same overwriting problem in S3 that this ticket is addressing.

So let's prevent that scenario by generating a completely random string, rather than depending on the epoch.

Reply:
💯